> pandas/core/frame.py:8298: error: List item 0 has incompatible type "Union[str, Tuple[Any, ...], List[Union[str, Tuple[Any, ...]]]]"; expected "Union[str, Tuple[Any, ...]]"  [list-item]

otherwise lgtm

--- a/pandas/core/frame.py
+++ b/pandas/core/frame.py
@@ -8201,7 +8201,7 @@ NaN 12.3   33.0
 
     def explode(
         self,
-        column: str | tuple | list[str | tuple],
+        column: Scalar | tuple | list[Scalar | tuple],
         ignore_index: bool = False,
     ) -> DataFrame:
         """
@@ -8211,7 +8211,7 @@ NaN 12.3   33.0
 
         Parameters
         ----------
-        column : str or tuple or list thereof
+        column : Scalar or tuple or list thereof
             Column(s) to explode.
             For multiple columns, specify a non-empty list with each element
             be str or tuple, and all specified columns their list-like data

--- a/pandas/core/frame.py
+++ b/pandas/core/frame.py
@@ -8293,7 +8293,7 @@ NaN 12.3   33.0
         if not self.columns.is_unique:
             raise ValueError("columns must be unique")
 
-        columns: list[str | tuple]
+        columns: list[Scalar | tuple]
         if is_scalar(column) or isinstance(column, tuple):
             columns = [column]
         elif isinstance(column, list) and all(

